### PR TITLE
KYLIN-3620 resolve "--" is treated as comment marker between singe qu…

### PR DESCRIPTION
**Pull request by caolijun1166**
KYLIN-3620 resolve "--" is treated as comment marker between singe quotes

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: query/src/main/java/org/apache/kylin/query/util/QueryUtil.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public static boolean isSelectStatement(String sql) {
 
     public static String removeCommentInSql(String sql1) {
         // match two patterns, one is "-- comment", the other is "/* comment */"
-        final String[] commentPatterns = new String[] { "--[^\r\n]*", "/\\*[\\s\\S]*?\\*/" };
+        final String[] commentPatterns = new String[]{"--(?!.*\\*/).*?[\r\n]", "/\\*(.|\r|\n)*?\\*/"};
```
Comment (shaofengshi) [inline query/src/main/java/org/apache/kylin/query/util/QueryUtil.java]: Is the code formatted?

File: query/src/test/java/org/apache/kylin/query/util/QueryUtilTest.java [hunk 2]
```diff
@@ -223,9 +223,22 @@ public void testRemoveCommentInSql() {
...
+
+        String content = "        --  One-line comment and /**range\n" +
+                "/*\n" +
+                "Multi-line comment\r\n" +
+                "--  Multi-line comment*/\n" +
+                "select price as " +
+                "/*\n" +
+                "Multi-line comment\r\n" +
+                "--  Multi-line comment*/\n" +
+                "revenue from /*One-line comment-- One-line comment*/ v_lineitem;";
```
Comment (shaofengshi) [inline query/src/test/java/org/apache/kylin/query/util/QueryUtilTest.java]: I didn't see the test case covers the situation of "--" in a quoted string.

File: query/src/main/java/org/apache/kylin/query/util/QueryUtil.java (hunk 1 above)
Comment (caolijun1166) [reply query/src/main/java/org/apache/kylin/query/util/QueryUtil.java]: Yes, it had been formatted.

**Pull request merged**